crates/cxx-qt-lib/src/std_types.cpp

```diff
@@ -0,0 +1,42 @@
+// clang-format off
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// clang-format on
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+#include <QtGlobal>
+
+// For versions less than Qt 6 we need to manually register the std numerics
+#if (QT_VERSION < QT_VERSION_CHECK(6, 0, 0))
+#include <QtCore/QMetaType>
+
+#include <cstdint>
+
+namespace rust {
+namespace cxxqtlib1 {
+
+// Ensure that std int types are registered
+// so that they can be used with QML in Qt 5
+
+static const int register_i8 =
```

Comment:
this causes the extension plugin tests to fail.

Is this the same problem as the `+whole-archive` option where things are compiled out as they aren't used ?

Eg this `std_types.cpp` doesn't get executed ...

Reply:
Moving the extension plugin example to use WHOLE_ARCHIVE in the CMake like other examples fixes this...